### PR TITLE
WIP: add snapshots for mocked and unmocked components

### PR DESCRIPTION
**Pull request by bcarroll22**
## Summary

Per a conversation with @TheSavior, in #24538, this adds snapshot tests for all components whose mocks will be addressed in that PR. Shallow and deep snapshots are included.

## Changelog

[General] [Added] - Snapshots

## Test Plan

This PR adds a lot of tests 😄

## Checklist

- [x] ActivityIndicator
- [x] DatePickerIOS
- [x] DrawerLayoutAndroid
- [x] Image
- [x] InputAccessoryView
- [ ] MaskedViewIOS
- [ ] Modal
- [ ] Picker
- [ ] PickerIOS
- [ ] ProgressBarAndroid
- [ ] ProgressViewIOS
- [ ] RefreshControl
- [ ] SafeAreaView
- [ ] ScrollView
- [ ] SegmentedControlIOS
- [ ] Slider
- [ ] StatusBar
- [ ] Switch
- [ ] Text
- [ ] TextInput
- [ ] ToolbarAndroid
- [ ] TouchableHighlight
- [ ] TouchableNativeFeedback
- [ ] TouchableOpacity
- [ ] TouchableWithoutFeedback
- [ ] View

### CODE REVIEW

**Comment by TheSavior** [comment]
We probably shouldn't do this all in one PR since it will be huge. It is totally reasonable (and probably preferred) to do these in chunks.

**Comment by bcarroll22** [comment]
👍 how do you want to see it split?

**Comment by TheSavior** [comment]
This is great, can you include the test for displayName too? We've messed that up in the past and having that test for every component would be great.

**Comment by TheSavior** [comment]
> 👍 how do you want to see it split?

The couple that you have in this PR right now is fine, but we probably shouldn't add more to this PR. Just small enough to easily review confidently, doesn't necessarily need to be one per PR. I leave it up to you to decide where the boundary should be.

**Comment by bcarroll22** [comment]
So for example, DatePickerIOS doesn't have a displayName. Should I add them if they aren't there?

File: Libraries/Components/DrawerAndroid/__tests__/__snapshots__/DrawerAndroid-test.js.snap [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+          "top": 0,
+        },
+        Object {
+          "backgroundColor": "white",
+          "width": undefined,
+        },
+      ]
+    }
+  >
+    <View />
```
Comment (TheSavior) [inline Libraries/Components/DrawerAndroid/__tests__/__snapshots__/DrawerAndroid-test.js.snap]: These are strange... 
These tests are going to be nice to be able to track these things down and know if we ever unintentionally change things

File: Libraries/Components/DrawerAndroid/__tests__/DrawerAndroid-test.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+ *
+ * @format
+ * @emails oncall+react_native
+ * @flow
+ */
+
+'use strict';
+
+const React = require('React');
+const AndroidDrawerLayout = require('../DrawerLayoutAndroid.android');
```
[the remark this answers is not in the document]
Comment (TheSavior) [reply Libraries/Components/DrawerAndroid/__tests__/DrawerAndroid-test.js]: I don't think you need to include the `.android`. Just `require('../DrawerLayoutAndroid')`

[the remark this answers is not in the document]
Comment (bcarroll22) [reply Libraries/Components/DrawerAndroid/__tests__/DrawerAndroid-test.js]: For some reason, in the tests, the platform was `ios` and I couldn't get it to change. So, before I did this, every render was an `UnimplementedView`

[the remark this answers is not in the document]
Comment (bcarroll22) [reply Libraries/Components/DrawerAndroid/__tests__/DrawerAndroid-test.js]: `jest.mock('Platform', () => ({ OS: 'android' })` wasn't working but this did

**Comment by TheSavior** [comment]
> So for example, DatePickerIOS doesn't have a displayName. Should I add them if they aren't there?

Hmm. It shouldn't be necessary to add them. I think it might actually be covered by `should shallow render as <DatePickerIOS> when not mocked`. If the display name wasn't set / parsed correctly by React then the snapshot test wouldn't include DatePickerIOS. When we ran into this problem was when we were converting class based components to function based components. I'm assuming that snapshot test will ensure we don't break the display name during that conversion. In which case, no need to add an additional test for displayName.

File: Libraries/Components/DrawerAndroid/__tests__/DrawerAndroid-test.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+ *
+ * @format
+ * @emails oncall+react_native
+ * @flow
+ */
+
+'use strict';
+
+const React = require('React');
+const DrawerLayoutAndroid = require('../DrawerLayoutAndroid.android');
```
[the remark this answers is not in the document]
Comment (bcarroll22) [reply Libraries/Components/DrawerAndroid/__tests__/DrawerAndroid-test.js]: 🤷‍♂️i dunno what to tell it on this one.

[the remark this answers is not in the document]
Comment (TheSavior) [reply Libraries/Components/DrawerAndroid/__tests__/DrawerAndroid-test.js]: This might be a question for @cpojer then

**Pull request closed without merging**